paper/tex/methods.tex

```diff
@@ -170,74 +171,97 @@ \subsubsection{GPz}
+% At a higher level, the pipeline mainly consists of three modules: (i) \textit{data pre-processing}, including a catalogue cross-matching sub-module \citep[based on the tool C3, ][]{Riccio:17}, a sub-module for photometric evaluation and error estimation of the multi-band catalogue used as Knowledge Base (KB), and a sub-module dedicated to the perturbation of the photometric KB, propaedeutic to the PDF estimation; (ii) \textit{photo-$z$ prediction}, which is the training/validation/test phase, producing the photo-$z$'s point estimates, based on a pre-selected ML method; (iii) \textit{PDF estimation}, specifically designed to calculate the PDF of the photo-$z$ estimation errors.
+% The last module includes also a post-processing tool, providing some statistics on the produced point estimates and PDFs.
 
+In this work, we used a hierarchical KNN to replace nondetections with values based on their neighbors.
+The usual cross-validation step was also omitted for this study.
 
 \subsubsection{SkyNet}
 \label{sec:skynet}
 %(J. Cohen-Tanugi and Hugo Tranin)
 
-\textsc{SkyNet}\footnote{\url{http://ccpforge.cse.rl.ac.uk/gf/project/skynet/}} \citep{Graff:14} is a publicly available neural network software, based on a 2nd order conjugate gradient optimization scheme \citep[see][for further details]{Graff:14}. %It has been used efficiently for redshift PDF estimates \citep{Sanchez:14,Bonnett:15,Bonnett:16}.
-
-%In the present work, we use \textsc{SkyNet} both as a regressor (for redshift point estimates) and a classifier (for PDF estimates).
-The neural network is configured as a standard multilayer perceptron with three hidden layers and one input layer with $12$ nodes (the $6$ magnitudes and their errors).
-%For the regressor, the output layer is a single linear node, while each hidden layer has $10$ nodes with a tanh activation function.
-The classifier is laid out such that the hidden layers have 20:40:40 nodes each, all rectified linear units, and the output layer has $200$ nodes (corresponding to 200 bins for the PDF) activated with a ``softmax'' function so that they automatically sum to $1$.  While previous implementations of the code, such as \citet{Sanchez:14} and \citet{Bonnett:15} (see Appendix C.3), implement a ``sliding bin'' smoothing, no such procedure was used in this study.
+\skynet \footnote{\url{http://ccpforge.cse.rl.ac.uk/gf/project/skynet/}} \citep{Graff:14} employs a neural network based on a second order conjugate gradient optimization scheme \citep[see][for further details]{Graff:14}. %It has been used efficiently for redshift PDF estimates \citep{Sanchez:14,Bonnett:15,Bonnett:16}.
+The neural network is configured as a standard multilayer perceptron with three hidden layers and one input layer with 12 nodes corresponding to the 6 photometric magnitudes and their measurement errors.
+We use \skynet\ as a regressor for \pz\ point estimation and as a classifier for \pzpdf\ estimation.
```

Comment:
Take out the text on as a regressor, as we discarded the point estimate stuff in the final paper.